lib/index.js

```diff
@@ -12,34 +26,308 @@ class PluginPassportOAuth {
+            Object.keys(this.config.strategies[profile.provider].kuzzleAttributesMapping)
+              .forEach(attr => {
+                req.body.content[attr] = profile._json[this.config.strategies[profile.provider].kuzzleAttributesMapping[attr]];
+              });
+          }
+
+          user._id = profile._json[this.config.strategies[profile.provider].identifierAttribute];
+          req.body.credentials[profile.provider] = user;
+
+          return this.context.accessors.execute(new this.context.constructors.Request(request.original, req))
+            .then(() => {
+              return Promise.resolve(profile._json[this.config.strategies[profile.provider].identifierAttribute]);
+            });
+        }
+        return Promise.resolve(profile._json[this.config.strategies[profile.provider].identifierAttribute]);
+      })
+      .catch(err => Promise.reject(err));
+  }
+
+  /**
+   * Check if a given user exists
+   *
+   * @param request
+   * @param userId
+   * @param provider
+   * @returns {Promise|Promise.<boolean>}
+   */
+  exists (request, userId, provider) {
+    return this.getCredentialsFromUserId(userId, provider)
+      .then(credentials => Promise.resolve(credentials !== null));
+  }
+
+  /**
+   * Create a user in the repository of the corresponding strategy
+   *
+   * @param request
+   * @param credentials
+   * @param userId user id in Kuzzle
+   * @param provider
+   * @returns {Promise|Promise.<TResult>}
+   */
+  create (request, credentials, userId, provider) {
+    let user = {};
+
+    return this.exists(request, userId, provider)
+      .then(exists => {
+        if (exists) {
+          return Promise.reject(new this.context.errors.BadRequestError('A strategy already exists for this user.'));
+        }
+        Object.keys(credentials).forEach(attr => {
+          user[attr] = credentials[attr];
+        });
+        user.userId = userId;
+
+        return this.getProviderRepository(provider).create(user);
+      });
+  }
+
+  /**
+   * Update a user in the repository of the corresponding strategy
+   *
+   * @param request
+   * @param credentials
+   * @param userId
+   * @param provider
+   * @returns {Promise|Promise.<TResult>}
+   */
+  update (request, credentials, userId, provider) {
+    return this.getCredentialsFromUserId(userId, provider)
+      .then(document => {
+        if (document === null) {
+          return Promise.reject(new this.context.errors.BadRequestError('A strategy does not exist for this user.'));
+        }
+
+        return this.getProviderRepository(provider).update(
+          Object.assign({_id: document._id}, credentials)
+        );
+      });
+  }
+
+  /**
+   * Delete a user in the repository of the corresponding strategy
+   *
+   * @param request
+   * @param userId
+   * @param provider
+   * @returns {Promise|Promise.<TResult>}
+   */
+  delete (request, userId, provider) {
+    return this.getCredentialsFromUserId(userId, provider)
```

Comment:
`userId` here is the user id in Kuzzle, so we should just call `this.getProviderRepository(provider).delete(userId, {refresh: 'wait_for'})`